### DISCUSSION TITLE
M1 - V8 engine initialisation crashes silently - logs?

### DISCUSSION THREAD

**Discussion by Mrazator** [ClearFoundry/ClearScript#373] [Q&A] [answered]
Initialisation of the V8 engine on M1 - `new V8ScriptEngine(constraints, flags, debugPort);` crashes silently during execution of the following method `V8SplitProxyNative.Impl_OSX_Arm64.V8Context_ExecuteCode`.

Is it possible to somehow get the reason of the crash? Is this stored inside some log?

```
.NET SDK (reflecting any global.json):
 Version:   6.0.202
 Commit:    f8a55617d2

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  12.0
 OS Platform: Darwin
 RID:         osx.12-arm64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.202/

Host (useful for support):
  Version: 6.0.4
  Commit:  be98e88c76

.NET SDKs installed:
  6.0.202 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.4 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

```
	  <PackageReference Include="Microsoft.ClearScript.V8" Version="7.2.4" />
	  <PackageReference Include="Microsoft.ClearScript.V8.ICUData" Version="7.2.4" />
          <PackageReference Include="Microsoft.ClearScript.V8.Native.osx-arm64" Version="7.2.4" />
```

**Comment by ClearScriptLib** [comment, marked as answer]
Hi @Mrazator,

At first glance, this looks like a case of Issue #318, so our first suggestion would be to try the procedure outlined [here](https://github.com/microsoft/ClearScript/issues/318#issuecomment-966831720).

To be honest, we aren't exactly macOS experts here, but you should be able to find crash reports in the [Console](https://support.apple.com/guide/console/welcome/mac) app.

Good luck!

**Reply by Mrazator** [reply]
Thanks for quick reply. In the Console app I don't see any logs, but fortunately this adhoc-signing made the crash go away. Thanks a lot!
